server/src/main/java/com/octopus/teamcity/opentelemetry/server/helpers/HelperPerBuildOTELHelperFactory.java

```diff
@@ -0,0 +1,75 @@
+package com.octopus.teamcity.opentelemetry.server.helpers;
+
+import jetbrains.buildServer.log.Loggers;
+import jetbrains.buildServer.serverSide.BuildPromotion;
+import jetbrains.buildServer.serverSide.ProjectManager;
+import jetbrains.buildServer.serverSide.SBuildServer;
+import jetbrains.buildServer.serverSide.crypt.EncryptUtil;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+
+import static com.octopus.teamcity.opentelemetry.common.PluginConstants.*;
+
+public class HelperPerBuildOTELHelperFactory implements OTELHelperFactory {
+    private final Map<Long, OTELHelper> otelHelpers;
+    private final ProjectManager projectManager;
+    private final SBuildServer sBuildServer;
+
+    public HelperPerBuildOTELHelperFactory(
+            ProjectManager projectManager,
+            SBuildServer sBuildServer
+    ) {
+        this.projectManager = projectManager;
+        this.sBuildServer = sBuildServer;
+        Loggers.SERVER.debug("OTEL_PLUGIN: Creating HelperPerBuildOTELHelperFactory.");
+
+        this.otelHelpers = Collections.synchronizedMap(new HashMap<>());
```

Comment:
I think I need this, but it'd be good to make sure we're both on the same page here @casstait - is `concurrentHashMap` better?

Reply:
They both have their tradeoffs. 
`ConcurrentHashMap` I prefer because you can iterate read AND write an object on different threads. I.e. its thread safe without having to lock the entire map. Only writes are done with a really finer grain lock (at a key/value pair level) which means it has better performance. Also it doesn't allow null keys or values which is nice. 
IF however performance isn't an issue and you want to ensure data consistency then use a `synchronizedMap`. The major difference here is that the `synchronizedMap` locks the entire object for both read and write operations. This essentially means only one thread at a time can access the object and blocks all other threads which may cause contention.

Reply:
I'd use a `ConcurrentHashMap` here tbh